### PR TITLE
[Jetcaster] Bump to alpha10 snapshot

### PR DESCRIPTION
**Pull request by yrezgui**
Addresses #339 issue

### CODE REVIEW

**Comment by manuelvicnt** [comment]
Is this ready to be reviewed?

File: Jetcaster/app/src/main/java/com/example/jetcaster/data/PodcastsRepository.kt [hunk 1]
```diff
@@ -43,8 +43,9 @@ class PodcastsRepository(
             refreshingJob?.join()
         } else if (force || podcastStore.isEmpty()) {
             refreshingJob = scope.launch {
+                try {
                 // Now fetch the podcasts, and add each to each store
-                podcastsFetcher(SampleFeeds).collect { (podcast, episodes, categories) ->
+                podcastsFetcher(SampleFeeds).collect{  (podcast, episodes, categories) ->
```
Comment (manuelvicnt) [inline Jetcaster/app/src/main/java/com/example/jetcaster/data/PodcastsRepository.kt:48]: Run the code formatting tool here. This line needs more indentation and a space should go between `collect` and the square bracket

File: Jetcaster/app/src/main/java/com/example/jetcaster/data/PodcastsRepository.kt [hunk 2]
```diff
@@ -60,6 +61,9 @@ class PodcastsRepository(
                         }
                     }
                 }
+                } catch (e: Throwable) {
+                    println("podcastsFetcher(SampleFeeds).collect error: $e")
```
Comment (manuelvicnt) [inline Jetcaster/app/src/main/java/com/example/jetcaster/data/PodcastsRepository.kt:65]: Can we do better than this? If not, let's use at least `Log.d` to print the message

File: Jetcaster/app/src/main/java/com/example/jetcaster/ui/JetcasterApp.kt [hunk 3]
```diff
@@ -16,11 +16,55 @@
...
+import com.example.jetcaster.R
 import com.example.jetcaster.ui.home.Home
 
+
 @Composable
 fun JetcasterApp() {
+    val context = AmbientContext.current
+    val isOnline = remember { mutableStateOf(checkIfOnline(context)) }
+
+    fun retry() {
```
Comment (manuelvicnt) [inline Jetcaster/app/src/main/java/com/example/jetcaster/ui/JetcasterApp.kt:38]: This function shouldn't be inside this composable. Even more, doesn't seem like it's used

File: Jetcaster/app/src/main/java/com/example/jetcaster/ui/JetcasterApp.kt [hunk 4]
```diff
@@ -16,11 +16,55 @@
...
+import androidx.compose.ui.platform.AmbientContext
+import androidx.compose.ui.res.stringResource
+import com.example.jetcaster.R
 import com.example.jetcaster.ui.home.Home
 
+
 @Composable
 fun JetcasterApp() {
+    val context = AmbientContext.current
+    val isOnline = remember { mutableStateOf(checkIfOnline(context)) }
```
Comment (manuelvicnt) [inline Jetcaster/app/src/main/java/com/example/jetcaster/ui/JetcasterApp.kt:36]: using `by` avoids having to use the `.value` below which reads better

```suggestion
    var isOnline by remember { mutableStateOf(checkIfOnline(context)) }
```

File: Jetcaster/app/src/main/java/com/example/jetcaster/ui/JetcasterApp.kt [hunk 5]
```diff
@@ -16,11 +16,55 @@
...
+        onDismissRequest = {},
+        title = { Text(text = stringResource(R.string.connection_error_title)) },
+        text = { Text(text = stringResource(R.string.connection_error_message)) },
+        confirmButton = {
+            TextButton(onClick = onRetry) {
+                Text(stringResource(R.string.retry_label))
+            }
+        }
+    )
+}
```
Comment (manuelvicnt) [inline Jetcaster/app/src/main/java/com/example/jetcaster/ui/JetcasterApp.kt:70]: New line missing at the end of this file. ktlint should've caught it

File: Jetcaster/app/src/main/java/com/example/jetcaster/ui/JetcasterApp.kt [hunk 6]
```diff
@@ -16,11 +16,55 @@
...
+    if(isOnline.value) {
+        Home()
+    } else {
+        OfflineDialog { isOnline.value = checkIfOnline(context) }
+    }
 }
+
+// TODO: Use a better way to check internet connection
+@Suppress("DEPRECATION")
+fun checkIfOnline(context: Context): Boolean {
```
Comment (manuelvicnt) [inline Jetcaster/app/src/main/java/com/example/jetcaster/ui/JetcasterApp.kt:52]: This function and the one below should be private

File: Jetcaster/app/src/main/java/com/example/jetcaster/data/PodcastsRepository.kt (hunk 2 above)
Comment (yrezgui) [reply Jetcaster/app/src/main/java/com/example/jetcaster/data/PodcastsRepository.kt:65]: It needs more UI work that will be pushed after. For now we have a dialog when the app is opened

**Review by manuelvicnt — APPROVED** [review]

File: Jetcaster/app/src/main/AndroidManifest.xml [hunk 7]
```diff
@@ -18,6 +18,7 @@
     package="com.example.jetcaster">
 
     <!-- Uses INTERNET to fetch RSS feed + images -->
+    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
```
Comment (chrisbanes) [inline Jetcaster/app/src/main/AndroidManifest.xml:21]: add a comment to why this is needed

**Review by chrisbanes — APPROVED** [review]
Just one small comment

**Pull request merged**